src/components/SubNavigation/SubNavigation.tsx

```diff
@@ -0,0 +1,179 @@
+    // shareData
+    toggleMiniTocOpen,
+    closeMiniToc,
+}: SubNavigationProps) => {
+    const [visibility, setVisibility] = useState(true);
+    const [hiddingTimeout, setHiddingTimeout] = useState<number | undefined>(undefined);
+    const [lastScrollY, setLastScrollY] = useState(window.screenY);
+
+    const shareData = useMemo(() => {
+        return {
+            title,
+            // url: window.location.href // ?
+        };
+    }, [title]);
+
+    const clickOutsideMiniToc = useCallback(
+        (event: MouseEvent) => {
+            /*
+             * func "composedPath" returns an array in which the last two elements are "HTML" and "#document",
+             * which do not have the classList property, so they are subtracted before checking by slice()
+             */
+            const isOutside = !event
+                .composedPath()
+                .slice(0, -2)
+                .some((item) => {
+                    const el = item as HTMLElement;
+                    const classes = el.classList ?? [];
+
+                    return classes?.contains('dc-doc-layout__right');
+                });
+
+            if (isOutside) {
+                closeMiniToc();
+            }
+        },
+        [closeMiniToc],
+    );
+
+    const controlVisibility = useCallback(() => {
+        if (miniTocOpened) {
+            setVisibility(true);
+            return;
+        }
+
+        if (lastScrollY === 0) {
+            setVisibility(true);
+        }
+
+        if (window.scrollY > lastScrollY) {
+            if (hiddingTimeout) {
+                return;
+            }
+
+            setVisibility(false);
+
+            setHiddingTimeout(
+                window.setTimeout(() => {
+                    clearTimeout(hiddingTimeout);
+                    setHiddingTimeout(undefined);
+                }, 300),
+            );
+        } else if (window.scrollY < lastScrollY) {
+            setVisibility(true);
+        }
+
+        setLastScrollY(window.scrollY);
+    }, [
+        miniTocOpened,
+        lastScrollY,
+        hiddingTimeout,
+        setLastScrollY,
+        setVisibility,
+        setHiddingTimeout,
+    ]);
+
+    const shareHandler = useCallback(() => {
+        if (navigator && navigator.share) {
+            navigator
+                .share(shareData)
+                .then(() => {})
+                .catch((error) => console.error('Error sharing', error));
+        } else {
+            console.log('Share not supported', shareData); // ?
+        }
+    }, [shareData]);
+
+    useEffect(() => {
+        if (window.scrollY === 0) {
+            return;
+        }
+
+        setHiddingTimeout(
+            window.setTimeout(() => {
+                setHiddingTimeout(undefined);
+            }, 100),
+        );
+    }, []);
+
+    useEffect(() => {
+        window.addEventListener('scroll', controlVisibility);
+
+        return () => {
+            window.removeEventListener('scroll', controlVisibility);
```

Comment:
It will not work, you need to memo the function in the scope like this. The code should have the actual value.
```
const fn = controlVisibility;
window.addEventListener('scroll', fn);

         return () => {
                      window.removeEventListener('scroll', fn);
```